### PR TITLE
Documentation for the auto proxy injection.

### PR DESCRIPTION
**Pull request by ihcsim**
This PR contains the documentation for the auto proxy injection feature. It includes instructions to install the proxy injector, testing the auto proxy injection and its current limitations.

I'm not sure if I'm doing this doc correctly as I am unable to run the hugo server locally, following the steps in the README.

### CODE REVIEW

**Comment by wmorgan** [comment]
Thanks @ihcsim!!!

File: linkerd.io/content/2/proxy-injection/_index.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+$ kubectl -n unlabeled run nginx-completed --image=nginx --port=80 --labels="linkerd.io/auto-inject=completed"
+deployment.apps "nginx-completed" created
+
+$ kubectl -n unlabeled get deploy
+NAME              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+nginx-completed   1         1         1            1           2m
+nginx-disabled    1         1         1            1           3m
+
+$ kubectl -n unlabeled get po -L linkerd.io/auto-inject
```
Comment (klingerf) [inline linkerd.io/content/2/proxy-injection/_index.md]: Whoa! TIL the `-L` flag -- so awesome.

**Comment by ihcsim** [comment]
@klingerf Should be all tidied up. When I tried to run `make server-linkerd.io`, I ended up with
```
$ make serve-linkerd.io
cp -R *linkerd.io tmp/
for fname in $(grep -rnl 'L5D2_STABLE_VERSION' tmp); do \
        sed -i 's/L5D2_STABLE_VERSION/stable-2.0.0/g' $fname; \
done
for fname in $(grep -rnl 'L5D2_EDGE_VERSION' tmp); do \
        sed -i 's/L5D2_EDGE_VERSION/edge-18.10.1/g' $fname; \
done
cd tmp/linkerd.io && ./build
ERROR: 2018/10/09 19:50:01 template.go:350: template: partials/meta.html:8: function "readFile" not defined
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/service-discovery.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/namerd.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/examples/external-resources.md".
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 shortcode.go:533: error processing shortcode shortcodes/sectiontoc.html
 ERR: template: shortcodes/sectiontoc.html:4:48: executing "shortcodes/sectiontoc.html" at <$currentPage.IsAnces...>: IsAncestor is not a field of struct type *hugolib.Page
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/load-balancing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/namerd.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/support/contact.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/deployment.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/transparent-proxying.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/load-balancing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/service-discovery.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/routing.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/features/tls.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/administration/telemetry.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/advanced/dtabs.md".
ERROR: 2018/10/09 19:50:01 template_funcs.go:1155: No page found with path or logical name "/1/administration/dtab-playground.md".
^CMakefile:63: recipe for target 'build-linkerd.io' failed
make: *** [build-linkerd.io] Interrupt
```
Not sure what I'm missing. FWIW, I did have to change the [`sed`](https://github.com/linkerd/website/blob/f4b4b144977e6506c140767f4bc8dcf5f36e0471/Makefile#L91) since I'm running on Linux.

**Comment by ihcsim** [comment]
@grampelberg Let me know if this is sufficient:
```
This feature is **experimental** and it's only available in the [_edge_ release](../edge/).
```
I have the tendency to get long-winded; trying to avoid that here :-)

**Comment by grampelberg** [comment]
@ihcsim yup, perfect. Thank you!

**Comment by grampelberg** [comment]
@ihcsim if I had to make a bet, your output looks like you might be on an older version of hugo. It is working with 0.48 for sure.

**Pull request merged**